---
title: "mod 13"
output:
  html_document: default
  pdf_document: default
date: "2024-04-23"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


```{r}
library(ggplot2)
library(dplyr)
data <- data.frame(
  MutationType = c(rep("Substitutions", 18), rep("Indels", 18)),
  Genotype = rep(c(rep("msh1", 9), rep("radA", 9), rep("Wild Type", 9)), 2),
  Genome = rep(c(rep("Mito", 9), rep("Plastid", 9)), 6),
  Replicate = rep(c(rep(1:3, 6)), 2),
  MutationRate = c(
    1.29E-06, 1.20E-06, 1.93E-06, 1.49E-07, 2.65E-07, 1.45E-07, 6.50E-08, 3.36E-08, 6.10E-08,
    3.45E-06, 5.88E-06, 4.15E-06, 5.05E-08, 3.78E-08, 3.70E-08, 1.00E-08, 6.90E-09, 5.00E-09,
    3.38E-06, 3.17E-06, 2.71E-06, 1.76E-07, 1.77E-07, 1.76E-07, 4.33E-08, 1.12E-08, 1.69E-08,
    1.33E-05, 1.73E-05, 1.47E-05, 1.12E-08, 6.05E-08, 3.70E-08, 5.63E-08, 6.90E-09, 1.00E-08
  )
)

data$Genotype <- factor(data$Genotype, levels = c("msh1", "radA", "Wild Type"))

data

avg_mutation_rates <- data %>%
  group_by(Genotype, Genome, MutationType) %>%
  summarize(AvgMutationRate = mean(MutationRate), .groups = "drop") 


print(avg_mutation_rates)


ggplot(data, aes(x = Genotype, y = MutationRate, fill = MutationType)) +
  geom_boxplot() +
  facet_grid(Genome ~ ., scales = "free_y", 
             space = "free_y") +
  labs(
    title = "Mutation Rates in Arabidopsis Mitochondrial and Chloroplast Genomes",
    x = "Genotype",
    y = "Mutation Rate (log scale)",
    fill = "Mutation Type"
  ) +
  theme_minimal() +
  scale_y_log10() + 
  theme(
    plot.title = element_text(size = 10, face = "bold"),
    axis.title = element_text(size = 14),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.text = element_text(size = 12),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 10)
  )


```